### PR TITLE
fix: add assistive labels to toggle filter button

### PR DESCRIPTION
**Pull request by tuanddd**
## Problem
The state of the button was not announced by the screen reader

## Solution
Add the assistive attributes of that like of a switch for the toggle filter button, where `checked=true` is equivalent to the filters being shown and `checked=false` is equivalent to the filters hidden

-----
SF-528

### CODE REVIEW

File: src/interface/Options/ToggleFilters.tsx [hunk 1]
```diff
@@ -10,7 +10,15 @@ export default () => {
   const { t } = useTranslation('filter');
 
   return (
-    <Button type="button" size="sm" onClick={toggleFilters} css={tw`whitespace-nowrap m-0`}>
+    <Button
+      role="checkbox"
+      aria-checked={filtersShown}
+      aria-label="Toggle Filters"
```
Comment (tuanddd) [inline src/interface/Options/ToggleFilters.tsx:16]: Would be ideal to have i18n in this attribute too but I'd rather put that in another PR.

Comment (zlatanpham) [reply src/interface/Options/ToggleFilters.tsx:16]: It's because the change requires high effort or something else?

File: src/interface/Options/ToggleFilters.tsx [hunk 2]
```diff
@@ -10,7 +10,15 @@ export default () => {
   const { t } = useTranslation('filter');
 
   return (
-    <Button type="button" size="sm" onClick={toggleFilters} css={tw`whitespace-nowrap m-0`}>
+    <Button
+      role="checkbox"
+      aria-checked={filtersShown}
```
Comment (zlatanpham) [inline src/interface/Options/ToggleFilters.tsx:15]: I reckon the attribute only makes sense for an element with checkbox role. Not sure if it can achieve what we wanted with a button.

File: src/interface/Options/ToggleFilters.tsx (hunk 1 above)
Comment (tuanddd) [reply src/interface/Options/ToggleFilters.tsx:16]: Yes, it would require a change in SDK and another in this repo to grab the new SDK version

File: src/interface/Options/ToggleFilters.tsx (hunk 2 above)
Comment (tuanddd) [reply src/interface/Options/ToggleFilters.tsx:15]: What can you suggest?

Comment (zlatanpham) [reply src/interface/Options/ToggleFilters.tsx:15]: Set `role="checkbox"` for the button 🧠

Comment (zlatanpham) [reply src/interface/Options/ToggleFilters.tsx:15]: Or can we toggle the value of `aria-label` between `Hide filters` and `Show filters` @tuanddd?

File: src/interface/Options/ToggleFilters.tsx [hunk 3]
```diff
@@ -8,10 +8,11 @@ export default () => {
   const { filtersShown, setFiltersShown } = useInterfaceContext();
   const toggleFilters = () => setFiltersShown(!filtersShown);
   const { t } = useTranslation('filter');
+  const label = filtersShown ? t('hide') : t('show');
```
Comment (zlatanpham) [inline src/interface/Options/ToggleFilters.tsx:11]: Should be `hide filters` and `show filters`, otherwise, it makes no difference to the button content.

```suggestion
  const label = filtersShown ? t('hide filters') : t('show filters');
```

**Review by wwalser — DISMISSED** [review]

File: src/interface/Options/ToggleFilters.tsx [hunk 4]
```diff
@@ -8,10 +8,22 @@ export default () => {
...
+      aria-checked={filtersShown}
+      type="button"
+      size="sm"
+      onClick={toggleFilters}
+      css={tw`whitespace-nowrap m-0`}
+      id="toggle-filters"
+    >
+      {label}
+      <Label css={tw`sr-only`} htmlFor="toggle-filters">
+        Toggle Filters
```
Comment (JasonBerry) [inline src/interface/Options/ToggleFilters.tsx:25]: We should add a translation for this:
```suggestion
        {t('toggle-filters')}
```

**Review by JasonBerry — APPROVED** [review]
Happy to merge, but we really should add a translation.

File: src/interface/Options/ToggleFilters.tsx (hunk 4 above)
Comment (tuanddd) [reply src/interface/Options/ToggleFilters.tsx:25]: Submitted https://github.com/sajari/sdk-react/pull/799 👍

**Review by JasonBerry — APPROVED** [review]

**Comment by tuanddd** [comment]
@JasonBerry Just a rebase

**Review by JasonBerry — APPROVED** [review]
